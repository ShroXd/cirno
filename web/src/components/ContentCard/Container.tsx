import { ReactNode } from 'react'

import { Card } from '@material-tailwind/react'

interface ContentCardContainerProps {
  children: ReactNode
  className?: string
}

export const ContentCardContainer = ({
  children,
  className,
}: ContentCardContainerProps) => (
  <Card
    className={`flex select-none flex-col overflow-hidden rounded-xl bg-white shadow-lg ${className || ''}`}
  >
    {children}
  </Card>
)
