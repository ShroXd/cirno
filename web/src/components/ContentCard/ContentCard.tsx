import { Typography } from '@material-tailwind/react'
import { FC } from 'react'

interface CardProps {
  imageUrl: string
  title: string
}

export const ContentCard: FC<CardProps> = ({ imageUrl, title }) => (
  <div className='bg-white flex flex-col select-none overflow-hidden shadow-lg rounded-xl'>
    <div className='overflow-hidden'>
      <img
        src={imageUrl}
        alt={title}
        className='w-full h-64 object-cover hover:scale-105 transition-all duration-700 ease-[linear(0,_0.402_7.4%,_0.711_15.3%,_0.929_23.7%,_1.008_28.2%,_1.067_33%,_1.099_36.9%,_1.12_41%,_1.13_45.4%,_1.13_50.1%,_1.111_58.5%,_1.019_83.2%,_1.004_91.3%,_1)]'
      />
    </div>
    <div className='px-4 py-3 bg-white'>
      <Typography
        variant='paragraph'
        className='text-center font-medium truncate'
      >
        {title}
      </Typography>
    </div>
  </div>
)
