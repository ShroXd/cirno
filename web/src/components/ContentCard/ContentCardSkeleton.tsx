import { PhotoIcon } from '@heroicons/react/24/outline'
import { CardBody, CardHeader, Typography } from '@material-tailwind/react'

import { ContentCardContainer } from './Container'

interface ContentCardSkeletonProps {
  className?: string
}

export const ContentCardSkeleton = ({
  className,
}: ContentCardSkeletonProps) => (
  <ContentCardContainer className={`animate-pulse ${className}`}>
    <CardHeader
      shadow={false}
      floated={false}
      className='relative m-0 mb-5 grid h-64 place-items-center rounded-b-none bg-gray-300'
    >
      <PhotoIcon className='h-12 w-12 text-gray-500' />
    </CardHeader>
    <CardBody className='pb-4 pt-0'>
      <Typography
        as='div'
        variant='h1'
        className='h-3 rounded-full bg-gray-300'
      >
        &nbsp;
      </Typography>
    </CardBody>
  </ContentCardContainer>
)
