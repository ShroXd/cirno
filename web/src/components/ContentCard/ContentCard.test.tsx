import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import { describe, expect, it } from 'vitest'

import { ContentCard } from './ContentCard'

describe('ContentCard', () => {
  it('should render without a11y violations', async () => {
    const { container } = render(<ContentCard imageUrl='' title='' />)
    expect(await axe(container)).toHaveNoViolations()
  })

  it('should render correctly with imageUrl, title', () => {
    const { container } = render(<ContentCard imageUrl='' title='Test title' />)

    expect(container).toBeVisible()
    expect(container.firstChild).toMatchSnapshot()
  })
})
