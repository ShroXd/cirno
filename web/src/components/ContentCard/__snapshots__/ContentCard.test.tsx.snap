// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContentCard > should render correctly with imageUrl, title 1`] = `
<div
  class="relative bg-clip-border text-gray-700 flex select-none flex-col overflow-hidden rounded-xl bg-white shadow-lg"
>
  <div
    class="overflow-hidden"
  >
    <img
      alt="Test title"
      class="h-64 w-full object-cover transition-all duration-700 ease-[linear(0,_0.402_7.4%,_0.711_15.3%,_0.929_23.7%,_1.008_28.2%,_1.067_33%,_1.099_36.9%,_1.12_41%,_1.13_45.4%,_1.13_50.1%,_1.111_58.5%,_1.019_83.2%,_1.004_91.3%,_1)] hover:scale-105"
      src=""
    />
  </div>
  <div
    class="bg-white px-4 py-3"
  >
    <p
      class="block antialiased font-sans text-base leading-relaxed text-inherit truncate text-center font-medium"
    >
      Test title
    </p>
  </div>
</div>
`;
